### PR TITLE
avc/vaapi_encode_h265: fix incorrect general_profile_compatibility_flag

### PR DESCRIPTION
**Pull request by feiwan1**
Follow spec Annex A.

Signed-off-by: Fei Wang <fei.w.wang@intel.com>

### CODE REVIEW

**Comment by xuguangxin** [comment]
could you give more details? like why we need to change this? 
Any error report by hm if we do not have this patch?

**Comment by feiwan1** [comment]
> could you give more details? like why we need to change this? Any error report by hm if we do not have this patch?

OK, will add more info into commit.

**Comment by feiwan1** [comment]
Hi all, any other comments?

**Comment by feiwan1** [comment]
Any other comments for this patch? If no, I am planning to send it to community. Thanks.

**Review by xuguangxin — APPROVED** [review]

File: libavcodec/vaapi_encode_h265.c [hunk 1]
```diff
@@ -299,11 +299,14 @@ static int vaapi_encode_h265_init_sequence_params(AVCodecContext *avctx)
     ptl->general_profile_idc   = avctx->profile;
     ptl->general_tier_flag     = priv->tier;
 
-    if (chroma_format == 1) {
-        ptl->general_profile_compatibility_flag[1] = bit_depth ==  8;
-        ptl->general_profile_compatibility_flag[2] = bit_depth <= 10;
+    ptl->general_profile_compatibility_flag[ptl->general_profile_idc] = 1;
+
+    if (ptl->general_profile_compatibility_flag[1])
```
Comment (xhaihao) [inline libavcodec/vaapi_encode_h265.c:304]: is it ptl->general_profile_compatibility_flag[1] in the condition ? The condition in the below if statement is ptl->general_profile_compatibility_flag[3], I guess here should be ptl->general_profile_compatibility_flag[2]

**Comment by xhaihao** [comment]
@feiwan1 I remember general_profile_idc(4) is PROFILE_HEVC_REXT,  is it not needed to handle ptl->general_profile_compatibility_flag[4] ?

File: libavcodec/vaapi_encode_h265.c (hunk 1 above)
Comment (feiwan1) [reply libavcodec/vaapi_encode_h265.c:304]: Should be general_profile_compatibility_flag[1]. Here is spec statement:

Conformance of a bitstream to the Main profile is indicated by general_profile_idc being equal to 1 or
general_profile_compatibility_flag[ 1 ] being equal to 1. Conformance of a sub-layer representation with TemporalId
equal to i to the Main profile is indicated by sub_layer_profile_idc[ i ] being equal to 1 or
sub_layer_profile_compatibility_flag[ i ][ 1 ] being equal to 1.
NOTE – When general_profile_compatibility_flag[ 1 ] is equal to 1, general_profile_compatibility_flag[ 2 ] should also be equal
to 1. When sub_layer_profile_compatibility_flag[ i ][ 1 ] is equal to 1 for a value of i, sub_layer_profile_compatibility_flag[ i ][ 2 ]
should also be equal to 1.

**Comment by feiwan1** [comment]
> 

According to spec, general_profile_compatibility_flag[general_profile_idc] must be 1 when general_profile_space is 0:

general_profile_compatibility_flag[ j ] equal to 1, when general_profile_space is equal to 0, indicates that the CVS
conforms to the profile indicated by general_profile_idc equal to j as specified in Annex A. When general_profile_space
is equal to 0, general_profile_compatibility_flag[ general_profile_idc ] shall be equal to 1. The value of
general_profile_compatibility_flag[ j ] shall be equal to 0 for any value of j that is not specified as an allowed value of
general_profile_idc in Annex A.

**Comment by xhaihao** [comment]
> > 
> 
> According to spec, general_profile_compatibility_flag[general_profile_idc] must be 1 when general_profile_space is 0:
> 
> general_profile_compatibility_flag[ j ] equal to 1, when general_profile_space is equal to 0, indicates that the CVS conforms to the profile indicated by general_profile_idc equal to j as specified in Annex A. When general_profile_space is equal to 0, general_profile_compatibility_flag[ general_profile_idc ] shall be equal to 1. The value of general_profile_compatibility_flag[ j ] shall be equal to 0 for any value of j that is not specified as an allowed value of general_profile_idc in Annex A.

So when general_profile_idc is 4, general_profile_compatibility_flag[4] should be 1, right ?

**Comment by feiwan1** [comment]
> general_profile_compatibility_flag

Yes.

**Comment by feiwan1** [comment]
> > > 
> > 
> > 
> > According to spec, general_profile_compatibility_flag[general_profile_idc] must be 1 when general_profile_space is 0:
> > general_profile_compatibility_flag[ j ] equal to 1, when general_profile_space is equal to 0, indicates that the CVS conforms to the profile indicated by general_profile_idc equal to j as specified in Annex A. When general_profile_space is equal to 0, general_profile_compatibility_flag[ general_profile_idc ] shall be equal to 1. The value of general_profile_compatibility_flag[ j ] shall be equal to 0 for any value of j that is not specified as an allowed value of general_profile_idc in Annex A.
> 
> So when general_profile_idc is 4, general_profile_compatibility_flag[4] should be 1, right ?

Yes.

**Comment by feiwan1** [comment]
Close this PR, and merge this to #469 which all fixed hevc encode issues.

**Pull request closed without merging**